src/Form.elm

```diff
@@ -197,14 +200,57 @@ update validation msg (F model) =
                     else
                         model.dirtyFields
 
-                newChangedFields =
-                    Set.insert name model.changedFields
+                ( newChangedFields, newOriginalValues ) =
+                    if Set.member name model.changedFields then
+                        let
+                            storedValue =
+                                Dict.get
+                                    name
+                                    model.originalValues
+                                    |> Maybe.withDefault Nothing
```

Comment:
Ok ok, well well, I totally forgot about reason why there is this part important :) Maybe I should add some comment here, but reason is that in originalValues is `Dict String (Maybe FieldValue)` for good reason (original value should be Nothing) and when I simply do `Dict.get name model.originalValues` I got `Maybe (Maybe FieldValue)` but for future processing I need only `Maybe FieldValue` and for nothing I need `Nothing` not `Just Nothing`. It should be rewritten to `Maybe.map` usage for better readability 🤔  

Something like that:
```
storedValue =
    Dict.get name model.originalValues

shouldBeNothing v =
    case v of
        Field.String "" ->
            True

        Field.Bool False ->
            True

        _ ->
            False

sameAsOriginal =
    Maybe.map (\sv -> 
    case sv of
        Just v ->
            v == fieldValue

        Nothing ->
            if shouldBeNothing fieldValue then
                True
            else
                False
    ) storedValue
```